feat: Move coverage submodule to repo DOCS-482 (#1526)
codacy/docs: docs/coverage-reporter/troubleshooting-coverage-cli-issues.md

---
description: Instructions or workarounds to overcome common issues while using Codacy Coverage Reporter CLI.
---

# Troubleshooting coverage CLI issues

The sections below provide instructions or workarounds to overcome common issues while using Codacy Coverage Reporter CLI.

## Commit SHA-1 hash detection {: id="commit-detection"}

The Codacy Coverage Reporter automatically detects the SHA-1 hash of the current commit to associate with the coverage data when you're using one of the following CI/CD platforms:

> -   Appveyor

Author: ```suggestion
-   <span class="skip-vale">Appveyor</span>
```

-   AWS CodeBuild
-   Azure Pipelines

> -   Bitrise

Author: ```suggestion
-   <span class="skip-vale">Bitrise</span>
```

> -   Buildkite

Author: ```suggestion
-   <span class="skip-vale">Buildkite</span>
```

-   Circle CI

> -   Codefresh

Author: ```suggestion
-   <span class="skip-vale">Codefresh</span>
```

-   Codeship
-   Docker
-   GitLab
-   Greenhouse CI
-   Heroku CI
-   Jenkins
-   Magnum CI
-   Semaphore CI
-   Shippable CI
-   Solano CI
-   TeamCity CI
-   Travis CI

> -   Wercker CI

Author: ```suggestion
-   <span class="skip-vale">Wercker CI</span>
```

> -   <span class="skip-vale">Wercker CI</span>

Reviewer: I'm thinking that instead of ignoring individual CI/CD platform names here, it could be best to just skip checking the entire section. :thinking: 

For instructing Vale to ignore multiple lines, there's a different syntax:

https://vale.sh/docs/topics/config/#markdown-amp-html

If the Codacy Coverage Reporter fails to detect the current commit from the CI workflow context, it will use the current commit from the local Git repository instead.

However, you can also force using a specific commit SHA-1 hash with the flag `--commit-uuid`. For example:

```bash
bash <(curl -Ls https://coverage.codacy.com/get.sh) report \
    -r report.xml \
    --commit-uuid cd4d000083a744cf1617d46af4ec108b79e06bed
```

## Can't guess any report due to no matching

Codacy Coverage Reporter automatically searches for coverage reports matching the [file name conventions for supported formats](index.md#generating-coverage).

> However, if Codacy Coverage Reporter does not find your coverage report, you can explicitly define the report file name with the flag `-r`. For example:

Author: ```suggestion
However, if Codacy Coverage Reporter doesn't find your coverage report, you can explicitly define the report file name with the flag `-r`. For example:
```

```bash
bash <(curl -Ls https://coverage.codacy.com/get.sh) report -r <coverage report file name>
```

## Report generated an empty result while uploading C# coverage data {: id="detailedxml"}

> If you're using dotCover to generate coverage reports for your C# projects, you must use the dotCover detailedXML report format as follows:

Author: ```suggestion
If you're using <span class="skip-vale">dotCover</span> to generate coverage reports for your C# projects, you must use the <span class="skip-vale">dotCover</span> detailedXML report format as follows:
```

```bash
dotCover.exe cover <...> --reportType=DetailedXml
```

## JsonParseException while uploading coverage data

If you get a `com.fasterxml.jackson.core.JsonParseException` error while uploading your coverage data to Codacy it means that your coverage report is too big and that Codacy Coverage Reporter hit a limit of 10 MB when uploading the coverage data to Codacy.

There are some ways you can solve this:

-   Split your coverage reports into smaller files and [upload them to Codacy one at a time](../uploading-coverage-in-advanced-scenarios/#multiple-reports).

> -   **If you're using dotCover to generate coverage reports for your C# projects**, you should [exclude xUnit files](https://www.jetbrains.com/help/dotcover/Running_Coverage_Analysis_from_the_Command_LIne.html#filters_cmd) from the coverage analysis as follows:

Author: ```suggestion
-   **If you're using dotCover to generate coverage reports for your C# projects**, you should [exclude <span class="skip-vale">xUnit</span> files](https://www.jetbrains.com/help/dotcover/Running_Coverage_Analysis_from_the_Command_LIne.html#filters_cmd) from the coverage analysis as follows:
```

    ```bash
    dotCover.exe cover <...> /Filters=-:xunit*
    ```

    By default, dotCover includes xUnit files in the coverage analysis and this results in larger coverage reports. This filter helps ensure that the resulting coverage data does not exceed the size limit accepted by the Codacy API when uploading the results.

## Connect timed out while uploading coverage data

If you get a `Error doing a post to <...> connect timed out` error while uploading your coverage data to Codacy it means that the Codacy Coverage Reporter is timing out while connecting to the Codacy API. This typically happens if you're uploading coverage data for larger repositories.

> To increase the default timeout while connecting to the Codacy API, use the flag `--http-timeout` to set a value larger than 10000 miliseconds. For example, to set the timeout to 30 seconds:

Author: ```suggestion
To increase the default timeout while connecting to the Codacy API, use the flag `--http-timeout` to set a value larger than 10000 milliseconds. For example, to set the timeout to 30 seconds:
```

```bash
bash <(curl -Ls https://coverage.codacy.com/get.sh) report \
    -r report.xml \
    --http-timeout 30000
```

## MalformedInputException while parsing report

> If you get a `java.nio.charset.MalformedInputException` when running the Codacy Coverage Reporter it means that the coverage report includes a character that is not encoded in UTF-8. The invalid character can belong to the file name of one of your source code files, or even a class or method name.

Author: ```suggestion
If you get a `java.nio.charset.MalformedInputException` when running the Codacy Coverage Reporter it means that the coverage report includes a character that's not encoded in UTF-8. The invalid character can belong to the file name of one of your source code files, or even a class or method name.
```

For maximum compatibility of your coverage reports with the Codacy Coverage Reporter, make sure that your coverage reports use UTF-8 encoding and that they only include UTF-8 characters.

## SubstrateSegfaultHandler caught signal 11

If you're experiencing segmentation faults when uploading the coverage results due to [oracle/graal#624](https://github.com/oracle/graal/issues/624), execute the following command before running the reporter, as a workaround:

```sh
echo "$(dig +short api.codacy.com | tail -n1) api.codacy.com" >> /etc/hosts
```

## coverage-xml/index.xml generated an empty result

If you're using PHPUnit version 5 or above to generate your coverage report, you must output the report using the Clover format. Codacy Coverage Reporter supports the PHPUnit XML format only for versions 4 and older.

To change the output format replace the flag `--coverage-xml <dir>` with `--coverage-clover <file>` when executing `phpunit`.

> See [PHPUnit command-line documentation](https://phpunit.readthedocs.io/en/9.5/textui.html) for more information.

Author: ```suggestion
See [<span class="skip-vale">PHPUnit</span> command-line documentation](https://phpunit.readthedocs.io/en/9.5/textui.html) for more information.
```

The Codacy Coverage Reporter automatically detects the SHA-1 hash of the current commit to associate with the coverage data when you're using one of the following CI/CD platforms:

> <!-- vale off -->

Author: @prcr Vale keeps complaining. Suggestions?

Reviewer: This is quite strange since the same syntax is working on other pages. :thinking: I tried running Vale locally and indeed it still detects the issues:

```bash
$ vale docs/*

 docs/coverage-reporter/uploading-coverage-in-advanced-scenarios.md
 66:41  error  Did you really mean 'golang'?  Vale.Spelling 

 docs/coverage-reporter/troubleshooting-coverage-cli-issues.md
 14:5    error  Did you really mean             Vale.Spelling 
                'Appveyor'?                                   
 17:5    error  Did you really mean 'Bitrise'?  Vale.Spelling 
 18:5    error  Did you really mean             Vale.Spelling 
                'Buildkite'?                                  
 20:5    error  Did you really mean             Vale.Spelling 
                'Codefresh'?                                  
 21:5    error  Did you really mean             Vale.Spelling 
                'Codeship'?                                   
 33:5    error  Did you really mean 'Wercker'?  Vale.Spelling 
 56:77   error  Did you really mean             Vale.Spelling 
                'detailedxml'?                                
 100:95  error  Did you really mean 'graal'?    Vale.Spelling 

 docs/coverage-reporter/index.md
 132:6   error  Did you really mean 'dariodf'?  Vale.Spelling 
 132:14  error  Did you really mean 'lcov'?     Vale.Spelling 
 133:6   error  Did you really mean             Vale.Spelling 
                'chrisgit'?                                   
 134:6   error  Did you really mean             Vale.Spelling 
                'danielpalme'?

[...]
```

Let's not waste time debugging Vale now (although, if you manage to replicate this consistently on a simpler Markdown file you could [open an issue on Vale's repository](https://github.com/errata-ai/vale/issues)), and revert to using the `<span class="skip-vale">...</span>` which seems to work. Sorry I made you lose some time around this!

Author: I did revert and open an issue. We'll see. https://github.com/errata-ai/vale/issues/529

Reviewer: Nice, thanks for opening the issue!
